gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -160,7 +162,7 @@ private void performRequest(
             addXForwardedHeaders(servletRequest, requestBuilder);
         }
 
-        ImmutableList<NewCookie> oauth2GatewayCookie = getOAuth2GatewayCookie(remoteUri, servletRequest);
+        List<NewCookie> cookies = getOAuth2GatewayCookie(remoteUri, servletRequest);
```

Comment:
Please use `ImmutableList.Builder` instead.

Reply:
This is intentionally a mutable list. Please check the code later which modifies the list for adding more cookies.

Reply:
I'm suggesting `ImmutableList.Builder` not `ImmutableList`